File: solutions/csharp/lucians-luscious-lasagna/LuciansLusciousLasagna.cs
class Lasagna
{
    private int PreparationTimePerLayer => 2;

    // Define the expected oven time in minutes
    public int ExpectedMinutesInOven() => 40;

    // Calculate the remaining oven time in minutes.
    public int RemainingMinutesInOven(int ElapsedBakeTime) => ExpectedMinutesInOven() - ElapsedBakeTime;

    // Calculate the preparation time in minutes.
    public int PreparationTimeInMinutes(int NumberOfLayers) => PreparationTimePerLayer * NumberOfLayers;

    // Calculate the elapsed time in minutes.
    public int ElapsedTimeInMinutes(int NumberOfLayers, int ElapsedBakeTime)
    {
        return PreparationTimeInMinutes(NumberOfLayers) + ElapsedBakeTime;
    }
}

File: solutions/fsharp/lucians-luscious-lasagna/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

// Define the expected oven time in minutes
let expectedMinutesInOven: int = 40

// Calculate the remaining oven time in minutes.
let remainingMinutesInOven (elapsedBakeTime: int) : int = expectedMinutesInOven - elapsedBakeTime

// Calculate the preparation time in minutes.
let preparationTimeInMinutes (numberOfLayers: int) : int =
    let preperationTimePerLayer: int = 2
    preperationTimePerLayer * numberOfLayers

// Calculate the elapsed time in minutes.
let elapsedTimeInMinutes (numberOfLayers: int) (elapsedBakeTime: int) : int =
    let prepTimeOfLayers: int = preparationTimeInMinutes numberOfLayers
    prepTimeOfLayers + elapsedBakeTime
